src/flash/display/MovieClip.ts

```diff
@@ -145,16 +171,36 @@ module Shumway.AVMX.AS.flash.display {
       var eventClass = this.sec.flash.events.Event.axClass;
       var queue: MovieClip[] = movieClipClass._callQueue;
       movieClipClass._callQueue = [];
+      var unsortedScripts: FrameScript [] = [];
+      
       for (var i = 0; i < queue.length; i++) {
         var instance = queue[i];
 
-        if (instance._hasFlags(DisplayObjectFlags.NeedsLoadEvent)) {
-          instance._removeFlags(DisplayObjectFlags.NeedsLoadEvent);
-          instance.dispatchEvent(eventClass.getInstance(events.Event.AVM1_LOAD));
+        if (movieClipClass.frameNavigationModel === FrameNavigationModel.SWF1) {
+          // AVM1 action blocks must be executed exactly in the same order they appear in the SWF
+          // file. We keep track of the position for each action block at parsing time and carry that
+          // information on to their wrapping functions (via the precedence property). Here we queue
+          // up all such functions so we can sort and execute them after this loop.
+          if (instance._hasFlags(DisplayObjectFlags.NeedsLoadEvent)) {
+            instance._removeFlags(DisplayObjectFlags.NeedsLoadEvent);
+            release || assert(instance._symbol);
+            var handler = (<AVM1.Lib.AVM1SymbolBase>
+              AVM1.Lib.getAVM1Object(instance, instance._symbol.avm1Context))._onLoadHandler;
```

Comment:
Okay, so I didn't fully understand how precedence is flowing. In this case we shall definitely remove _onLoadHandler and handling of precedence from AVM1 logic -- virtual machine shall not know about intricacies of timeline things. We can save precedence at placeobject (I don't think actionblock precedence matters in this case, worst case we need to get it from the placeobject events from actionsblock), and that will make much more sense.

I think, the _onLoadHandler hack will prevent us from handling onLoad events (http://help.adobe.com/en_US/AS2LCR/Flash_10.0/help.html?content=Part2_AS2_LangRef_1.html) that are bound dynamically.